node/src/daemon_json/json_genesis.rs

```diff
@@ -0,0 +1,66 @@
+use chrono::DateTime;
+use serde::{Deserialize, Serialize};
+
+use mina_p2p_messages::{
+    number::Number,
+    v2::{
+        BlockTimeTimeStableV1, MinaBaseProtocolConstantsCheckedValueStableV1,
+        UnsignedExtendedUInt32StableV1, UnsignedExtendedUInt64Int64ForVersionTagsStableV1,
+    },
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Genesis {
+    k: u32,
+    delta: u32,
+    slots_per_epoch: u32,
+    slots_per_sub_window: u32,
+    grace_period_slots: u32,
+    genesis_state_timestamp: String,
+}
+
+impl Genesis {
+    pub fn k(&self) -> UnsignedExtendedUInt32StableV1 {
+        UnsignedExtendedUInt32StableV1(Number(self.k))
+    }
+
+    pub fn delta(&self) -> UnsignedExtendedUInt32StableV1 {
+        UnsignedExtendedUInt32StableV1(Number(self.delta))
+    }
+
+    pub fn slots_per_epoch(&self) -> UnsignedExtendedUInt32StableV1 {
+        UnsignedExtendedUInt32StableV1(Number(self.slots_per_epoch))
+    }
+
+    pub fn slots_per_sub_window(&self) -> UnsignedExtendedUInt32StableV1 {
+        UnsignedExtendedUInt32StableV1(Number(self.slots_per_sub_window))
+    }
+
+    pub fn grace_period_slots(&self) -> UnsignedExtendedUInt32StableV1 {
+        UnsignedExtendedUInt32StableV1(Number(self.grace_period_slots))
+    }
+
+    pub fn genesis_state_timestamp(&self) -> anyhow::Result<BlockTimeTimeStableV1> {
```

Comment:
I wouldn't use `anyhow` based errors in library crates.

Reply:
Done in d3d4829f.